### PR TITLE
`color-function-notation`: add more criteria to the function that determines the notation

### LINKED ISSUE #7945: `color-function-notation`: do not attempt to autofix when the syntax is invalid
### What minimal example or steps are needed to reproduce the bug?

stylelint --fix changes ```a { color: rgb(white, .5) }``` to ```a { color: rgb(white .5) }```, which sass fails to compile.

### What minimal configuration is needed to reproduce the bug?

```json
{ "rules": { "color-function-notation": "modern" } }
```

### How did you run Stylelint?

```sh
pnpm exec stylelint path/to/file.scss -c path/to/stylelint.config.js --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "16.8.2",
    "stylelint-config-standard-scss": "13.1.0",
    "stylelint-order": "6.0.4",
    "stylelint-scss": "6.5.0",
}
```

### What did you expect to happen?

stylelint should suggest a valid syntax in this case (probably `rgb(white, 50%)`)

### What actually happened?

stylelint replaces `rgb(white, .5)` with `rgb(white .5)` which sass fails to compile

### Do you have a proposal to fix the bug?

_No response_

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

Closes #7945

> Is there anything in the PR that needs further explanation?

N/A

### CODE REVIEW

**Comment by ybiquitous** [comment]
> 1. No need for new tests because we are not officially supporting Sass.

I think adding a case like `rgb(white, 50%)` is still worth it because such an invalid CSS is not limited to Sass, for example, a user's misunderstanding etc.

File: lib/rules/color-function-notation/index.mjs [hunk 1]
```diff
@@ -142,11 +144,18 @@ function isComma(node) {
...
+	if (nodes.length < 5) return false;
+
+	let commas = 0;
+
+	for (const node of nodes) {
+		if (isComma(node)) commas++;
+		else if (node.value === '/') return false;
+	}
+
+	return commas && (commas === 2 || commas === 3);
```
Comment (ybiquitous) [inline lib/rules/color-function-notation/index.mjs:158]: [suggest] `commas` (equals `commas > 0`) seems unnecessary: 👀 

```suggestion
	return (commas === 2 || commas === 3);
```

Comment (Mouvedia) [reply lib/rules/color-function-notation/index.mjs:158]: That's an early return when you have a modern notation.
i.e. `=== 0`

**Comment by Mouvedia** [comment]
> > 1. No need for new tests because we are not officially supporting Sass.
> 
> I think adding a case like `rgb(white, 50%)` is still worth it because such an invalid CSS is not limited to Sass, for example, a user's misunderstanding etc.

I disagree. I think it sets a bad precedent.
i.e. we are only fixing regressions related to Sass/Less at this point
If the category of the rule was [invalid](https://stylelint.io/user-guide/rules/#invalid), I would agree with you.
This rule should do the strict minimum.
i.e. performance over perfection

Why should I add `a { color: rgba(,,,); }` to the tests?
What does it accomplish?

File: lib/rules/color-function-notation/index.mjs (hunk 1 above)
Comment (Mouvedia) [reply lib/rules/color-function-notation/index.mjs:156]: It's an early return so that you don't have to check the second condition.

**Comment by ybiquitous** [comment]
I don't think stricter test cases are required, but this PR causes some behavior change, so I think we still need some test cases to prove the behavior change.

**Review by ybiquitous — COMMENTED** [review]
Thanks, overall LGTM. Besides, can you add a changelog item? This PR fixes `color-function-notation` false positives.

File: lib/rules/color-function-notation/index.mjs [hunk 2]
```diff
@@ -67,9 +67,13 @@ const rule = (primary, secondaryOptions) => {
 
 				if (!LEGACY_NOTATION_FUNCTION_NAME.test(value)) return;
 
-				if (primary === 'modern' && !hasCommas(node)) return;
+				const hasEnoughArguments = nodes.length >= 5;
 
-				if (primary === 'legacy' && hasCommas(node)) return;
+				if (!hasEnoughArguments) return;
+
+				if (primary === 'modern' && !isLikelyLegacy(nodes)) return;
```
Comment (romainmenke) [inline lib/rules/color-function-notation/index.mjs:74]: Maybe assign the result of `isLikelyLegacy(nodes)` to a variable so that it can be reused?

Comment (romainmenke) [reply lib/rules/color-function-notation/index.mjs:74]: nvm :)

Only one is ever executed because `primary` is always `modern` or `legacy`.

**Review by ybiquitous — APPROVED** [review]
Thanks, LGTM 👍🏼

**Pull request merged**